compiler/rustc_const_eval/src/interpret/place.rs

```diff
@@ -646,9 +647,13 @@ where
         assert!(dest.layout().is_sized(), "Cannot write unsized immediate data");
 
         match self.as_mplace_or_mutable_local(&dest.to_place())? {
-            Right((local_val, local_layout)) => {
+            Right((local_val, local_layout, local)) => {
                 // Local can be updated in-place.
                 *local_val = src;
+                // Call the machine hook (the data race detector needs to know about this write).
+                if !self.validation_in_progress() {
```

Comment:
Are these checks for `valiation_in_progress()` just an optimization, or are they required for correctness?

Reply:
This matches the equivalent logic in `memory.rs`. There it is definitely required; we have some "administrative reads" that we don't want to show up in the aliasing model (specifically, the ones added in https://github.com/rust-lang/rust/pull/128942).

Here I am not sure if it is required, but it would seem odd to not make this consistent with the logic in `memory.rs`.